src/content/BatchItem.ts

```diff
@@ -0,0 +1,196 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module BatchItem
+ */
+
+/**
+ * @interface
+ * Signature represents key value pair object
+ */
+import {
+  Parsable,
+  ParseNode,
+  SerializationWriter,
+  UntypedNode,
+  createUntypedNodeFromDiscriminatorValue,
+  isUntypedString,
+  isUntypedBoolean,
+  isUntypedNull,
+  isUntypedNumber,
+  isUntypedArray,
+  isUntypedObject,
+} from "@microsoft/kiota-abstractions";
+
+/**
+ * @interface
+ * Signature represents payload structure for batch request and response
+ */
+export interface BatchItem {
+  readonly id: string;
+  method: string;
+  url: string;
+  headers?: Record<string, string> | null;
+  body?: Record<string, any> | null;
```

Comment:
What if the body is a stream?

Reply:
@rkodev please avoid resolving conversation until a consensus has been reached. It makes progress much harder to track reviewers' side.